schema/query.go

```diff
@@ -156,9 +156,13 @@ func validateQuery(q map[string]interface{}, validator Validator, parentKey stri
 			if parentKey == "" {
 				return nil, fmt.Errorf("%s can't be at first level", op)
 			}
-			n, ok := isNumber(exp)
-			if !ok {
-				return nil, fmt.Errorf("%s: value for %s must be a number", parentKey, op)
+			var v interface{}
+			if n, ok := isNumber(exp); ok {
+				v = n
+			} else if t, ok := isTime(exp); ok {
```

Comment:
You should only accept time condition if the validator is of type `Time` and number if of type `Float` or `Integer` (see on line 166 below).

Here you blindly accept time or number, and only on number fields and you refuse both on time fields.